src/main/java/org/czekalski/petapiintegration/controllers/AuthorizationController.java

```diff
@@ -1,61 +1,160 @@
 package org.czekalski.petapiintegration.controllers;
 
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.czekalski.petapiintegration.apiClient.v1.dto.AnimalDto;
+import org.czekalski.petapiintegration.apiClient.v1.dto.AnimalsListDto;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.security.core.Authentication;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.security.oauth2.client.OAuth2RestTemplate;
-import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.io.IOException;
+
 @RestController
 public class AuthorizationController {
 
-
-
     @Value("${messages.base-uri}")
     private String messagesBaseUri;
 
     @Autowired
-    private OAuth2RestTemplate messagingClientClientCredsRestTemplate;
+    private OAuth2RestTemplate auth2RestTemplatee;
```

Comment:
Zawsze powinieneś ustawiać dependencies przez konstruktor. Robi się to dlatego, że łatwiej jest testować takie klasy. Później podajesz jedynie w konstruktorze mock, stub albo fake. Gdy jest to prywatne pole, to już nie takie proste.
```suggestion
    private final OAuth2RestTemplate auth2RestTemplate;
    
    public AuthroizationController(OAuth2RestTemplate auth2RestTemplate) {
        this.restTemplate = restTemplate;
    }
```

Reply:
To jest wciąż nierozwiązane

Reply:
Ok, chociaż tę klasę skasowałem bo to była testowa